runbot_buildout/models/runbot_build.py

```diff
@@ -0,0 +1,335 @@
+# Copyright 2017-2018 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import glob
+import os
+import shutil
+import subprocess
+import tempfile
+from odoo import api, fields, models
+from odoo.tools import file_open
+from odoo.addons.runbot.common import lock, grep
+
+
+class RunbotBuild(models.Model):
+    _inherit = 'runbot.build'
+
+    @api.multi
+    def _server(self, *l, **kw):
+        self.ensure_one()
+        if self.repo_id.uses_buildout:
+            odoo_part = 'parts/%s' % self.repo_id.buildout_section
+            if os.path.exists(self._path(odoo_part, 'odoo')):
+                return self._path(odoo_part, 'odoo', *l, **kw)
+            return self._path(odoo_part, 'openerp', *l, **kw)
+        else:
+            return super(RunbotBuild, self)._server(*l, **kw)
+
+    @api.multi
+    def _cmd(self):
+        self.ensure_one()
+        if self.repo_id.uses_buildout:
+            if not os.path.exists(self._path('datadir')):
+                os.mkdir(self._path('datadir'))
+            cmd = [
+                self._path('bin/start_%s') % self.repo_id.buildout_section,
+                '--xmlrpc-port=%d' % self.port,
+                '--data-dir', self._path('datadir'),
+                '--addons-path', self._server('addons'),
+            ]
+            if grep(self._server('tools/config.py'), 'no-xmlrpcs'):
+                cmd.append('--no-xmlrpcs')
+            if grep(self._server('tools/config.py'), 'no-netrpc'):
+                cmd.append('--no-netrpc')
+            if grep(self._server('tools/config.py'), 'log-db'):
+                cmd.append('--log-db=%s' % self.env.cr.dbname)
+            return cmd, self.modules
+        else:
+            return super(RunbotBuild, self)._cmd()
+
+    @api.model
+    def _job_00_init(self, build, lock_path, log_path):
+        if build.repo_id.uses_buildout and\
+                not build.branch_id.buildout_version:
+            build._github_status()
+            buildout_build = build._get_buildout_build()
+            if not buildout_build:
+                return -2
+            build._log('buildout', 'Using buildout %s@%s' % (
+                buildout_build.branch_id.name,
+                buildout_build.name,
+            ))
+            if not build._init_from_buildout(buildout_build):
+                return -2
+            buildout_file = build._adapt_buildout()
+            if not buildout_file:
+                return -2
+            build._log('buildout', 'Running buildout')
+            return build._spawn_buildout(
+                [
+                    build._path('bin/buildout'), '-N', '-q',
+                    '-c', buildout_file,
+                ],
+                lock_path, log_path,
+            )
+        else:
+            return super(RunbotBuild, self)._job_00_init(
```

Comment:
if I have `build.repo_id.uses_buildout=True` but don't have `build.branch_id.buildout_version` should continue the normal runbot process?